rubicon/objc/objc.py

```diff
@@ -867,6 +873,40 @@ def __call__(self, receiver, *args, convert_args=True, convert_result=True):
                 result = ObjCClass(result)
             return result
 
+######################################################################
+
+class ObjCPartialMethod(object):
+    _sentinel = object()
+    
+    def __init__(self, name_start):
+        super().__init__()
+        
+        self.name_start = name_start
+        self.methods = {}
+    
+    def __repr__(self):
+        return "{cls.__module__}.{cls.__qualname__}({self.name_start!r})".format(cls=type(self), self=self)
+    
+    def __call__(self, receiver, first_arg=_sentinel, **kwargs):
+        if first_arg is ObjCPartialMethod._sentinel:
+            if kwargs:
+                raise TypeError("Missing first (positional) argument")
+            
+            args = []
+            rest = frozenset()
+        else:
+            args = [first_arg]
+            # Add "" to rest to indicate that the method takes arguments
```

Comment:
I see below it's being decoded for selectors.

Is this convention more general?